### ISSUE TITLE
resin build omits last boolean arguments if param [source] is present

### ISSUE THREAD

**Issue by jacintoArias** [balena-io/balena-cli#579]
I have been experiencing a weird behaviour with `resin build` that I've tracked to the following situation.

Suppose our current path contains a valid project and we call this:
```
resin build --deviceType raspberrypi3 --arch armhf --emulated --nocache 
```
then everything works as expected, but if we call
```
resin build --deviceType raspberrypi3 --arch armhf --emulated --nocache  .
```
then the build ignores the `--nocache` argument and builds using cache. I'm curious about this tool and the **capitano** project so I dived in the code of other different actions in resin cli but couldn't find any differences with `resin build`. 

I made however another comparison, I hope it could accelerate the process of finding the bug. If we add:
```
Object.keys(options).forEach (opt) ->
     logging.logInfo( logging.getLogStreams(), opt)
```
[on line 62 of](https://github.com/resin-io/resin-cli/blob/974be5cc13cb33ba6110fc5e9d6b5b51a9864b2f/lib/actions/build.coffee#L62) `lib/actions/build.coffee` and repeat the previous process you get the list of captured arguments in which the last one is missing, respectively:

```
[Info]    arch
[Info]    deviceType
[Info]    emulated
[Info]    nocache
```
and
```
[Info]    arch
[Info]    deviceType
[Info]    emulated
```

Note: the last argument is captured if the last one has a parameter, such as:
```
resin build --deviceType raspberrypi3 --emulated --nocache  --arch armhf 
```

**Comment by pimterry** [comment]
@jacintoArias It looks like this is because capitano parses it as a parameter for `nocache`, which is a boolean toggle, so that doesn't make any sense. I agree the result is confusing though, I'll take a look at it.

In the mean time, you can move the parameter to the start of the command (`resin build . --deviceType ...`) and I think everything'll work fine.

**Comment by pdcastro** [comment]
This year we started migrating from [Capitano](https://github.com/balena-io/capitano/) to [oclif](https://oclif.io/), one CLI command at a time, so now the plan to fix this issue is to migrate the `balena build` command to oclif.

**Comment by pdcastro** [comment]
Fixed in CLI release v12.9.9 (and later).
